.github/workflows/images-branch.yml

```diff
@@ -18,9 +18,35 @@ on:  # yamllint disable-line rule:truthy
         description: PR number
 
 jobs:
+  approval:
+    runs-on: ubuntu-latest
+    outputs:
+      decision: ${{ steps.decision.outputs.decision }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set PR
+        run: echo "PR=${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        if: github.event_name != 'workflow_dispatch'
+
+      - name: Set PR (forks)
+        run: echo "PR=${{ inputs.pr }}" >> $GITHUB_ENV
+        if: github.event_name == 'workflow_dispatch'
+
+      # Skips push-updater-images by setting an output
+      - name: Check if pull request is approved
+        id: decision
+        run: |
+          DECISION=$(gh pr view ${{ env.PR }} --json reviewDecision,state -t '{{.reviewDecision}}:{{.state}}')
+          echo "decision=$DECISION" >> $GITHUB_OUTPUT
+
   push-updater-images:
-    name: Deploy
     runs-on: ubuntu-latest
+    needs: approval
+    if: needs.approval.outputs.decision == 'APPROVED:OPEN' && !github.event.pull_request.head.repo.fork
```

Comment:
I found it very confusing that we checked for approval & also checked that not a fork, so once I puzzled through to understand why this was written this way, I was able to tidy it up a little more:
* https://github.com/dependabot/dependabot-core/pull/11212